From 5ebab2c43abd4204832bd6e7194f20f12dfed279 Mon Sep 17 00:00:00 2001
From: Javier Pena <jpena@redhat.com>
Date: Tue, 19 Feb 2019 19:18:10 +0100
Subject: [PATCH] Set proper python_package_name for RHEL/CentOS 8

mysql::bindings::python installs the package set by the
python_package_name variable, which is 'MySQL-python' by default
on all RHEL and Fedora-based systems. This is no longer valid for
RHEL 8, since there is no Python3 support for the package.

In RHEL 8, we should use python-PyMySQL instead.
---
 manifests/params.pp | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/manifests/params.pp b/manifests/params.pp
index b8def7d..ed6ff2c 100644
--- a/manifests/params.pp
+++ b/manifests/params.pp
@@ -49,6 +49,7 @@ class mysql::params {
           } else {
             $provider = 'mysql'
           }
+          $python_package_name = 'MySQL-python'
         }
         /^(RedHat|CentOS|Scientific|OracleLinux)$/: {
           if versioncmp($::operatingsystemmajrelease, '7') >= 0 {
@@ -56,6 +57,11 @@ class mysql::params {
           } else {
             $provider = 'mysql'
           }
+          if versioncmp($::operatingsystemmajrelease, '8') >= 0 {
+            $python_package_name = 'python3-PyMySQL'
+          } else {
+            $python_package_name = 'MySQL-python'
+          }
         }
         default: {
           $provider = 'mysql'
@@ -96,7 +102,6 @@ class mysql::params {
       $java_package_name       = 'mysql-connector-java'
       $perl_package_name       = 'perl-DBD-MySQL'
       $php_package_name        = 'php-mysql'
-      $python_package_name     = 'MySQL-python'
       $ruby_package_name       = 'ruby-mysql'
       $client_dev_package_name = undef
     }
-- 
2.20.1

